### ISSUE TITLE
Centered Layout

### ISSUE THREAD

**Issue by davidmerrique** [metafizzy/packery#8]
Are there any plans to add a "centered" option? Like the "Centered Masonry" in Isotope? 

http://isotope.metafizzy.co/custom-layout-modes/centered-masonry.html

That's the only thing that's missing for me right now.

**Comment by voltmn** [comment]
Thanks a lot for this brilliant work with packery script, it's magic :)  I have a question about isotope+packery centering. Is it possible to merge this js http://codepen.io/desandro/pen/chisC
and packery-mode.pkgd.min.js (not packery.pkgd.js) right? Because it's not working together.

**Comment by desandro** [comment]
@voltmn Sure thing. Use this CodePen: http://codepen.io/desandro/pen/wBxvKe

``` js
// overwrite Packery methods
var PackeryMode = Isotope.LayoutMode.modes.packery;
var __resetLayout = PackeryMode.prototype._resetLayout;
PackeryMode.prototype._resetLayout = function() {
  __resetLayout.call( this );
  // reset packer
  var parentSize = getSize( this.element.parentNode );
  var colW = this.columnWidth + this.gutter;
  this.fitWidth = Math.floor( ( parentSize.innerWidth + this.gutter ) / colW ) * colW;
  this.packer.width = this.fitWidth;
  this.packer.height = Number.POSITIVE_INFINITY;
  this.packer.reset();
};

PackeryMode.prototype._getContainerSize = function() {
  // remove empty space from fit width
  var emptyWidth = 0;
  for ( var i=0, len = this.packer.spaces.length; i < len; i++ ) {
    var space = this.packer.spaces[i];
    if ( space.y === 0 && space.height === Number.POSITIVE_INFINITY ) {
      emptyWidth += space.width;
    }
  }

  return {
    width: this.fitWidth - this.gutter,
    height: this.maxY - this.gutter
  };
};

// always resize
PackeryMode.prototype.needsResizeLayout = function() {
  return true;
};
```

**Comment by voltmn** [comment]
Wow, so easy! Thanks!

**Comment by voltmn** [comment]
@desandro Hello again! :) I've found that Isotope sets container's height to zero when the centring fix is in use. http://jsfiddle.net/u9xg8uvo/

**Comment by voltmn** [comment]
I think the problem was that I used this CodePen (http://codepen.io/desandro/pen/wBxvKe) for the packery-mode  v1.1.1, now I see that there is a new version 1.1.2. But this CodePen does not work in new version.

**Comment by Lucas-C** [comment]
I tried to use that hack, but it looks like `Packery.prototype.resize` does not call the `.needsResizeLayout() ` method defined in `Outlayer`. Hence, overriding `Packery.prototype.needsResizeLayout` as done is the CodePen does not work : the function simply does not get called.

As a result, the layout resize does not always happen because the `return` in `Packery.prototype.resize` acts as a shortcircuit.
I hope my explanations are clear enough. Do you see a way to solve this issue ?

Also, are you still considering to implement that behaviour as a functionality of Packery ? Because else such a hack will be difficult to maintain as Packery's code evolve.

**Comment by cyllu2** [comment]
Is it possibly to center the whole thing like for instance Pinterest does? With css it won't center very well, because if you resize the window and an element is moved somewhere else it will leave a big hole in the design.

**Comment by n8-sd** [comment]
> I could add isFitWidth feature and then add a disclaimer that it only works with fixed-sized columns. But I see this as more problematic than just removing the feature all together.

I disagree with this, a functioning integrated option that doesn't work in a situation thats in the documentation, thats fine.

Currently am using isotope via a vue wrapper and can't get you js solutions working via that.

**Comment by NBurke1** [comment]
+1

sometimes things just need to be centered :)

Struggling to get the codepen examples working, but I'm sure i'll get there eventually...

**Comment by xe4me** [comment]
Unfortunately, none of these examples is working properly.
As soon as you put a max-width in the container, it blows
